context.go

```diff
@@ -21,25 +21,61 @@
 package fx
 
 import (
-	gcontext "context"
+	"context"
 
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber/jaeger-client-go"
+	"go.uber.org/fx/service"
 	"go.uber.org/fx/ulog"
 )
 
-// Context embeds Host and Go stdlib context for use
-type Context interface {
-	gcontext.Context
+type contextKey int
 
-	Logger() ulog.Log
+const _contextStore contextKey = iota
+
+type ctxStore struct {
```

Comment:
context.Value() is a `o(n)` operation, with `n` being number of elements added `WithValue`. Keeping it in struct will save us future optimization if/when we will add more per request data like auth context. rpc team had good arguments in https://github.com/uber-go/fx/issues/126 and PR to keep the optimization in place, and I don't see any point not to.